core/frontend/src/components/vehiclesetup/MotorDetection.vue

```diff
@@ -0,0 +1,233 @@
+        <StatusTextWatcher :filter="/thrust|Thrust|Motor/" @message="updateStatus" />
+      </v-sheet>
+      <v-row v-if="status == statuses.IDLE">
+        <v-col rows="3" class="justify-center">
+          <ThemedSVG src="/public/img/icons/motordetection.svg" />
+        </v-col>
+        <v-col cols="8">
+          <v-card-text>
+            <p>
+              Tests whether any of the motors are wired to spin backwards,
+              and automatically reverses their control signals as relevant.
+            </p>
+            <p>
+              The test relies on the motors being correctly connected according to the selected vehicle frame, and
+              spins each motor individually to check whether it generates thrust in the expected direction.
+              The vehicle should be in calm water, and the autopilot must be in manual mode to be able to run the test.
+            </p>
+          </v-card-text>
+        </v-col>
+      </v-row>
+
+      <v-sheet v-if="[statuses.RUNNING].includes(status)" class=" pa-3 ma-8">
+        <v-progress-linear indeterminate />
+        The test is running. The vehicle will move slightly as each motor is tested.
+      </v-sheet>
+
+      <v-alert
+        v-if="status == statuses.COMPLETE"
+        class="ma-8"
+        dense
+        text
+        type="success"
+      >
+        Motor direction detection is complete. The vehicle is ready for normal operation.
+      </v-alert>
+      <v-alert
+        v-if="status == statuses.FAILED"
+        class="ma-8"
+        type="error"
+        dense
+        text
+      >
+        Motor direction detection failed. Please check the troubleshooting section below.
+      </v-alert>
+      <v-card-text v-if="status == statuses.FAILED" class="ma-8">
+        <p>
+          Motor testing can fail due to any unexpected motions of the vehicle. Here are some common failure causes,
+          and potential solutions:
+        </p>
+        <v-sheet class="elevation-2 pa-1 mt-3 mb-3 mr-9 rounded-lg">
+          <v-simple-table class="ma-0">
+            <thead>
+              <tr>
+                <th>Cause</th>
+                <th>Solutions</th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr>
+                <td>no movement</td>
+                <td>
+                  - ensure the vehicle is in the water<br>
+                  - ensure the ESC is powered and receiving valid signals<br>
+                  - ensure the thruster is connected to the ESC
+                </td>
+              </tr>
+              <tr>
+                <td>external movement</td>
+                <td>
+                  - ensure the vehicle is in calm water
+                </td>
+              </tr>
+              <tr>
+                <td>poor movement estimation</td>
+                <td>
+                  - ensure the sensors are well calibrated<br>
+                  - ensure the vehicle is not too close to large magnetic disturbances
```

Comment:
(metalic structures)

Reply:
oh that is actually wrong, cleaning it up